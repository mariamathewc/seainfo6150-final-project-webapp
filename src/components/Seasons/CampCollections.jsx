import React from "react";
import styles from "./CampCollections.module.css";
import CampList from "./CampList";
import { useLocation } from "react-router";

const CampCollections = (props) => {

    const location = useLocation();
    const selectedSeason = location.state.code; 
    console.log("test", location.state.campdetail[selectedSeason]);


    let displayContent;

    if (location.state.campdetail[selectedSeason].length) {

        displayContent = (

            <ul className={styles.container} >
                {location.state.campdetail[selectedSeason].map((camp) => (
                   
                    <CampList camp={camp} key={camp.title} />
                ))}
            </ul>

        );


    } else {
        displayContent = <div>You have no data!</div>;
    }

    return (
        <div >
            <div className={styles.cover}>
                
            </div>

            <div className={styles.textformat}>
                <h2>View our hand-picked campgrounds around the world</h2>
            </div>

            <div >

                {displayContent}
            </div>
        </div>			
		)
	
}

export default CampCollections 