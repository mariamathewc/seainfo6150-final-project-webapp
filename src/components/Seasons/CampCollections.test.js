import React from "react";
import CampCollections from "./CampCollections.jsx";
import { render } from "@testing-library/react";
import routeData from 'react-router';

const mockLocation = {
	pathname: 'localhost:3000/spring',

}


describe("Camp collection test", () => {
	it("renders correctly", () => {
		const { container } = jest.spyOn(routeData, 'useLocation').mockReturnValue(mockLocation);
		expect(container).toMatchSnapshot();
	});
});

