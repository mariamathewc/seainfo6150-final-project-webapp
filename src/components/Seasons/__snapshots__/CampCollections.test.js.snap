// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Camp collection test renders correctly 1`] = `undefined`;
